src/utils/__tests__/spotify.spec.js

```diff
@@ -0,0 +1,213 @@
+import axios from "axios";
+import { fetchTracks, fetchUser, requestToApi } from "../spotify";
+import RetriesCountExceededError from "../RetriesCountExceededError";
+
+jest.mock("axios");
+
+describe("Tests for fetchUser", () => {
+  const response200 = {
+    data: {
+      display_name: "A",
+      id: "123",
+      images: [{ url: "a.com" }],
+    },
+    status: 200,
+  };
+  const expected = {
+    username: "A",
+    id: "123",
+    profileImage: "a.com",
+    hasError: false,
+  };
+  const error500 = {
+    response: {
+      data: { message: "internal server error" },
+      status: 500,
+    },
+  };
+  const error429 = {
+    response: {
+      data: { message: "Rate limiting" },
+      status: 429,
+      headers: {
+        "retry-after": 2,
+      },
+    },
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it("should fetch user data", () => {
+    axios.mockResolvedValue(response200);
+
+    return fetchUser("token").then((data) => expect(data).toEqual(expected));
+  });
+
+  it("should retry request", async () => {
+    axios
+      .mockReturnValueOnce(new Promise((resolve, reject) => reject(error500)))
+      .mockReturnValueOnce(new Promise((resolve) => resolve(response200)));
+
+    const actual = await fetchUser("token");
+
+    expect(actual).toEqual(expected);
+    expect(axios.mock.calls.length).toBe(2);
+  }, 1500);
+
+  it("should retry request after specified time in retry-after header", async () => {
+    axios
+      .mockReturnValueOnce(new Promise((resolve, reject) => reject(error429)))
+      .mockReturnValueOnce(new Promise((resolve) => resolve(response200)));
+
+    const actual = await fetchUser("token");
+
+    expect(actual).toEqual(expected);
+
+    expect(axios.mock.calls.length).toBe(2);
+  }, 2500);
+
+  it("should throw RetriesCountExceededError when too much retries", async () => {
+    axios.mockRejectedValue(error500);
+
+    try {
+      await fetchUser("token");
+    } catch (e) {
+      expect(e).toEqual(
+        new RetriesCountExceededError("Count of retries is exceeded")
+      );
+    }
+  }, 15000);
```

Comment:
it's not clear why timeouts needed. it's bad when unit test needs to wait a lot. you can make delays configurable as function arguments (ie `requestToApi(url, token, retryCount=10, retryDelay=500)`), then in tests you can set 1ms delay and assert if wait for 3ms when you have 3 retries etc